docs/en/pact-reference.md

```diff
@@ -1277,9 +1277,200 @@ Declaring models shares the same syntax with modules:
+    }
+    (a-token::transfer a-account pair-a-account a-amount)
+    (b-token::transfer pair-b-account b-account b-amount))
+)
+```
+
+
+### Modrefs and Polymorphism
+
+Modrefs provide polymorphism for use cases like the example above with an emphasis on
+interoperability. A modref is specified with one or more interfaces, allowing
+for values of that modref to reference modules that implement those
+interfaces.
+
+In the calling example above, the modref `a-token:module{fungible-v2}` accepts a
+reference to the Kadena `coin` KDA token module, because `coin` implements
+`fungible-v2`. Of course there is nothing special about `fungible-v2`: modrefs
+can specify any defined interface and accept any module that implements said
+interface.
+
+In this way, modrefs bring polymorphism to Pact. To programmers coming from
+polymorphic languages like Javascript, Java or Python, module references may
+look familiar. However, Pact is very much not object-oriented so there are
+important differences in how polymorphism works and what it is best used for.
+
+A key difference is that while modrefs allow polymorphism over modules, they
+don't offer polymorphism over data. Modules in Pact are not data in any
+practical sense: not only are they "code only", but they cannot be dynamically
+created within the Pact system.
+
+Meanwhile, data stored in the database is managed using schemas, which have no
+polymorphic features via modrefs or any other mechanism. This is why we suggest
+modrefs are more about "interoperation" instead of polymorphism per se, avoiding
+the object-oriented connotation of "objects changing shape". Modrefs allow modules to
+"interoperate with each other".
+
+Additionally, programmers should resist the urge to employ other object-oriented
+patterns using modrefs. A popular idea of "coding around interfaces, not
+implementations" is actually harmful in Pact if polymorphism is not
+needed. Programmers should prefer direct references whenever possible as they
+are not only faster but safer, as Pact aggressively and permanently inlines all
+directly-referenced code in the interests of code stability.
+
+### Important concerns when using modrefs.
+
+#### Late Binding
+
+As of Pact 4.7, modrefs are "late-binding", which means that the latest
+upgraded version of a module will be used when a module operation is invoked.
+
+Consider a modref to a module stored in the database when the module is
+at version 1. Sometime later the module is upgraded to version 2. The modref
+in the database will refer to the upgraded version 2 of the module when read
+back in and used.
+
+As described in the [Dependency Management](#dependency-management) section,
+Pact direct references are not late-binding, so this modref behavior might
+be surprising. "Pinned modrefs" that bind eagerly to the module version is
+being actively considered as an upcoming Pact change to extend the benefits of
+dependency management to modrefs.
```

Comment:
Not sure if we want to provide an example of how one could `modref` to a pinned module in the docs, but one could create a simple wrapper like:

```pact
(module coin-v5-pin G
  (use coin) ; pin to latest coin or use hash
  (defcap G() (enforce false "Can not upgrade"))
  
  (defun get-coin:module{fungible-v2}() coin)) ; or implement fungible and wrap all modules 
```

Reply:
Dunno, maybe I should just take this out and we'll just ship pins in the next release and update the docs.